src/hooks/useSections.ts

```diff
@@ -15,20 +15,33 @@ export default function useSections(
   advancedParameters?: AdvancedParameters
 ) {
   const zeroStateActiveSections = !query.length && zeroStateSections;
-  const [activeSections, setActiveSections] = useState(
+
+  // Define All Sections
+  const [activeSections, setActiveSections] = useState<UserDefinedSection[]>(
     zeroStateActiveSections ? zeroStateSections : sections
   );
 
+  useEffect(() => {
+    setActiveSections(zeroStateActiveSections ? zeroStateSections : sections);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [zeroStateActiveSections]);
```

Comment:
I may be missing something, but this looks like derived state at this point. 
Can we simplify? 

all of this:
```javascript
  // Define All Sections
  const [activeSections, setActiveSections] = useState<UserDefinedSection[]>(
    zeroStateActiveSections ? zeroStateSections : sections
  );

  useEffect(() => {
    setActiveSections(zeroStateActiveSections ? zeroStateSections : sections);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [zeroStateActiveSections]);
```

to just, this:
this:
```javascript
  const activeSections = zeroStateActiveSections ? zeroStateSections : sections;
```

Reply:
I thought I wanted to make it reactive to only zeroStateActiveSections so it updates the state only if zeroStateActiveSections changes. But whenever zeroStateActiveSections, zeroStateSections and sections also change and they're passed as props to the hook so the whole hook rerun every time any way.

TLDR; I thought this was an optimization but I don't think it is anymore

Reply:
https://github.com/Constructor-io/constructorio-ui-autocomplete/pull/107/commits/55c168e54f3bc1568827cba615592e193f2fc3c1#diff-c72a6a39c2792562569f2b7e56003020f1eefbc541ba6221750d044635f0d1b1R20